### ISSUE TITLE
Use Tags to determine Notification 

### ISSUE THREAD

**Issue by derekoharrow** [louislam/uptime-kuma#5285]
### 📑 I have found these related issues/pull requests

I cannot find any related requests.

### 🏷️ Feature Request Type

Settings

### 🔖 Feature description

Tags seem to have changed in V2 - in V1 they use to have a tag and a value, now it just seems to be a tag.

This seems an ideal opportunity to start to use tags for other things.

In this instance, it would be great to have a Setting on a Notification that applies that notification to any monitor with a specific tag.

e.g. I could create a tag called "Notify Me", and then in my Pushover notification settings have that notification apply to any monitors with the Notify Me tag - this would save having to setup notifications on every monitor individually.

### ✔️ Solution

Add the ability to list tags against Notification agents so that those agents automatically apply to any monitor with those tags.

**Comment by CommanderStorm** [comment]
Tangling the notification and the Tag system seem like a really strange choice. I don't think that this is a good choice and would add a ton of issues from users not knowing where their notifications are coming from.

**Comment by CommanderStorm** [comment]
You are likely looking for config templates or bulk editing instead..

**Comment by derekoharrow** [comment]
> Tangling the notification and the Tag system seem like a really strange choice. I don't think that this is a good choice and would add a ton of issues from users not knowing where their notifications are coming from.

I disagree. If Tags cannot be used for things like this, then what is their value, other than cosmetic?

I have over 170 monitors and setting the notifications for them individually is painful to say the least. 

Even with a bulk editor you'd need to decide what notifications to enable for each and, like with Maintenance windows, ensure you have selected every monitor you want individually which is really prone to error. (In fact, Notification windows would be another good use of tags).

Using tags is the perfect solution and choosing good tag names such as "notify-p1" easily gets around the issue of knowing what triggers what. 

Coming from an Enterprise IT monitoring background, manually configuring each individual monitors notifications is painful at best and very error prone. Tags would be a simple and effective way to automate much of that.

**Comment by Ionys320** [comment]
I have a similar situation, where my services have different priorities. For example, our internal Weblate instance is not as necessary as our production platform (used by our users). Therefore, we set different notifications methods for both of them (Weblate sends a mail, Prod send Mail + Discord ping). We have 10 services like this, so it's pretty simple to manage them that way, but I think being capable to bind a notification to a tag could be useful.

**Comment by capo42** [comment]
We try to replace PRTG with uptime kuma. There are only a few small concerns that holds us back from doing so.
On PRTG we simply add a tag to a sensor. Let say notify-247 or notify-59. And assign the tags to different notification targets.
So we would really appreciate the notification to a tag feature!

**Comment by craig-halliday** [comment]
**+1 on this.** We're using tags to mark monitors as `personal-device` (phones, PCs, tablets) and want to exclude them from push notifications without manually managing per-monitor notification links. The ability to either:
- **Include** a notification for all monitors *with* a specific tag, or  
- **Exclude** a notification for monitors *with* a specific tag  

would cover a lot of real-world use cases cleanly. Right now the only workaround is inserting directly into the `monitor_notification` table and hoping a future UI change doesn't wipe it.
